webdriver-spec.html

```diff
@@ -375,20 +395,75 @@ <h4 id='returning-http-response'>Returning a HTTP response to the local end</h4>
-  <h2>Sessions</h2>
-  <p>Non-normative summary: A session is equivalent to a single instantiation of a particular browser, including all
-  child windows. The WebDriver API gives each session a unique ID stored as an opaque string that can be used to differentiate one
-  session from another, allowing multiple browsers to be controlled on the same machine if needed, and allowing
-  sessions to be routed via a multiplexer. </p>
-  <p>The following definitions relate to this section:</p>
-  <ul>
-    <li>The session id is stored in the <dfn id='#sessionId' title="sessionid">sessionId</dfn> field.</li>
-    <li>The <dfn id='capabilities'>Capabilities</dfn> object described throughout the specification is
-        the underlying collection of capabilities that a User Agent handle, represented as a dictionary
-        mapping strings to values that can be serialised
-              The default value is an empty dictionary.
-    </li>
-  </ul>
+
+<h2>Sessions</h2>
+
+<p><em>Non-normative summary:</em>
+ A session is equivalent to a single instantiation of a particular user agent,
+ including all its child browsers.
+ WebDriver gives each session a unique <a>session ID</a>
+ that can be used to differentiate one session from another,
+ allowing multiple user agents to be controlled on the same machine if needed,
+ and allowing sessions to be routed via a multiplexer.
+
+<p>A WebDriver <dfn>session</dfn> represents
+ the connection between a local end and a specific remote end.
+ A remote end that is not an <a>intermediary node</a>
+ must have at most one <a>active session</a> at a given time.
+<!-- Is this actually a requirement? -->
+
+<!--
+ TODO(jgraham): Need to explain that there must be
+ a relationship between the current session
+ and the browser state.
+-->
+
+<p>The <a>session</a> is set up at the invocation of
+ a <a href=#newsession>new session</a>,
+ and torn down at some later point;
+ either explicitly by invoking <a href=#>quit</a>,
+ or implicitly when <a href=#close>close</a> is called at the last remaining
+ <a href=http://www.w3.org/TR/html5/browsers.html#top-level-browsing-context>top level browsing context</a>.
+
+<p>Each <a>remote end</a> has an associated <dfn>list of active sessions</dfn>,
+ which is a list of all <a rel="session">sessions</a> that are currently started.
+
+<p>The <a>remote end</a> has an associated <a>current session</a>
+ that is the <a rel="active session">session that is active</a>
+ for the duration of a <a>command</a>.
+
+<p>A <a>session</a> has an associated <dfn>session ID</dfn> (a <a>UUID</a>)
+ used to uniquely identify thi session.
+ Unless stated otherwise it is null.
+
+<p>A <a>session</a> has an associated <dfn>session capabilities</dfn>
+ object (a <a>capabilities</a> object)
+ describing the functionality provided by this <a>session</a>.
+ Unless stated otherwise it is null.
+
+<p>A <a>session</a> has an associated <dfn>current browsing context</dfn>
```

Comment:
Can we give this a name without "current" in it? Because I think you will often want to say "the |current session|'s |current browsing context|", which sounds weird. "active" is quite close to what HTML uses.